src/JPEGView/AVIFWrapper.cpp

```diff
@@ -93,9 +95,22 @@ void* AvifReader::ReadImage(int& width,
 	if (cache.transform == NULL)
 		cache.transform = ICCProfileTransform::CreateTransform(icc.data, icc.size, ICCProfileTransform::FORMAT_BGRA);
 	ICCProfileTransform::DoTransform(cache.transform, cache.rgb.pixels, cache.rgb.pixels, width, height);
+
+	avifRWData exif = cache.decoder->image->exif;
+	if (exif.size > 8 && exif.size < 65528 && exif.data != NULL) {
+		exif_chunk = malloc(exif.size + 10);
+		if (exif_chunk != NULL) {
+			memcpy(exif_chunk, "\xFF\xE1\0\0Exif\0\0", 10);
+			*((unsigned short*)exif_chunk + 1) = _byteswap_ushort(exif.size + 8);
+			memcpy((uint8_t*)exif_chunk + 10, exif.data, exif.size);
+		}
+	}
+
+	void* pPixelData = cache.rgb.pixels;
 	if (!has_animation)
 		DeleteCache();
-	return cache.rgb.pixels;
+
+	return pPixelData;
```

Comment:
I noticed the comment about fixing a leak, where was the leak?  Is this just a cast of `cache.rgb.pixels` to `void*`?

Reply:
If an AVIF wasn't animated then `DeleteCache()` would set `cache.rgb.pixels` to NULL without being freed. I didn't notice because we use libheif as a backup if libavif fails.